streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -145,6 +147,26 @@ public StreamTask(TaskId id,
     }
 
     /**
+     * Finds the source of the given topic partition.
+     *
+     * @param partition the topic partition
+     * @return
+     */
+    private SourceNode findSource(TopicPartition partition) {
+        SourceNode source = topology.source(partition.topic());
+        // if the source is defined by pattern, we check the sources one by one
+        if (source == null) {
+            for (String topic: topology.sourceTopics()) {
```

Comment:
Why not use `topology.sourceTopicPattern()` ? And than check if `partition.topic()` matches the pattern?